### DISCUSSION TITLE
Statistical error as shapeStat

### DISCUSSION THREAD

**Discussion by jakobnov** [histfitter/histfitter#95] [Q&A]
Dear HistFitter @histfitter/developers and experts,

I've come across an issue with the treatment of the statistical error in the HistFitter, which I think originates from the HistFactory. Namely, in our analysis, we have a data-driven background which has a floating normalisation factor in the fit. If I treat the statistical error in the same way as for other MC samples:

`fakes.setStatConfig(True)`,

the final exclusion limit depends on the initial normalisation of the data-driven background. For example, if I scale the normalisation of the input background distribution by a factor smaller than 1, the final limit gets lower, and vice versa. The Asimov data is kept the same, so the final expected values of the background sample are not affected, only the normalisation factor is rescaled. I've read the HistFactory [documentation](https://cds.cern.ch/record/1456844/files/CERN-OPEN-2012-016.pdf) and it seems to me that the expected value of the background sample contains dependency on the nuisance parameters, but its standard deviation doesn't (section 2.2.1). Which means that the absolute stat. error of a particular sample is kept constant, while its normalisation can be a floating parameter.

I wanted to workaround this by disabling stat. errors and manually add Poissonian terms to the likelihood, to make sure that the relative error is kept constant (not the absolute error). From the list of systematic methods in HistFitter, it seems to me that `shapeStat` has the right functionality for that. This code summarises what I did:

```
statSys = Systematic("statError","Nom","dummy","dummy","tree","shapeStat","Gaussian")
fakes.setStatConfig(False)
fakes.addSystematic(statSys)
```

If I understand correctly, the variation of the systematic is read from the stat. errors of the bins of the input histogram. I use `Gaussian` constraint, since the fit with `Poisson` constraint is very time consuming. Could you, please, confirm that this approach in fact adds Gaussian (Poissonian) terms for gamma factors into the likelihood? And that the variation of the gamma factors corresponds to the relative stat. error of my input histogram and is kept constant in the fit?

Thanks,
Jakob

**Comment by matthewfeickert** [comment]
In addition to the @histfitter/developers, @longjon929 might be able to give some guidance here.

**Comment by luilongo** [comment]
Dear Jacob,  
HistFactory creates one gamma nuissance parameter which controls the total statistical error obtained from all the samples for which `setStatConfig` is set to `true`. The statistical error is therefore not set constant but it’s included in the fit through this gamma parameter. `shapeStat` implements also a gamma nuissance parameter but this time per sample. If the sample has a normalization factor, during the fit the correlation between the normalization factor and the gamma factor will be estimated and, in principle, even a bigger initial stat error can contribute less to the total error thanks to the possible anticorrelation with  the scale factor. 
There is still something that I don’t understand from your question. If you are using asimov data you should get a normalization factor equal to 1 during the fit; what do you mean by scaling the normalization? Could you prepare some examples? 
Thanks,
 Luigi

**Reply by jakobnov** [reply]
Hi Luigi,

thanks for your reply!

Let me clarify the procedure. First I perform a background-only fit in CRs. The Asimov data in the SR is constructed from the background samples, weighted by the normalisation factors from the background-only fit. So they are equal to input factors by definition when fitting on blinded data.

I meant the following, with the rescaled normalisation factor: I arficially reduced my background normalisation (and its stat. error) by 0.6. But I still used the nominal histogram values, reweighted by the normalisation factor in the Asimov data. The normalisation factor from the background-only fit is 0.6 as well, which means that in the signal region fit I will obtain normalisation 1, by definition. I was surprised that this affects the limit. I would expect that rescaling of the background shouldn't affect the limit, as long as it's normalisation is a floating parameter of the fit.

Ok, I think your answer confirms that `shapeStat` is indeed a suitable workarond of the issue. Can you confirm, that the machinery running behind the scenes is somewhat different for `shapeStat` and `setStatConfig(True)`? Because according to my tests, using `setStatConfig(True)` doesn't take into account stat. error correlation with the normalisation factor.

Thanks,
Jakob

**Reply by luilongo** [reply]
Hi Jakob,
if I've understood correctly you have always the same Asimov data but you are changing your bkg normalization. With no rescaling you will have the correct scale factor at 0.6, while rescaling you will have a scale factor equal to 1. If I think to a simple Poissonian without the nuissance parameter I would have expected the same result, maybe you could try to run without any systematic and see if this is still the case.
Both `shapeStat` and `setStatConfig(True)` are based on the HistFactory `shapeSys` and they are the same thing. The only difference is that ShapeStat is applied to a specific sample and not to the full bkg of the channel. Therefore I still don't understand exactly the problem. 
Could you write here how you are adding the normfactor to your sample? Could you also run in verbose mode HF for the 2 cases? In the verbose mode it's explicitly shown the product of the weights.
Let me know,
Luigi

**Reply by jakobnov** [reply]
Hi Luigi,

I might have an idea of what could be the issue. Do I get it right that the relative stat. error of all the background samples is treated as contant in the fit, when using `setStatConfig(True)`? My data-driven background (fakes) is subleading in terms of the expected value, since dominant background of our analysis is ttbar. But the stat. error of the fakes is comparable to the stat. error of the ttbar. Which means that the normalisation of the background prediction is almost intact by rescaling of the fakes, but the stat. error is affected, hence the difference in the relative error.

I assume the correct treatment would be to separate the fakes from the rest of the MC backgrounds. And if I understand correctly how `shapeStat` works, this is achieved precisely by:

`statSys = Systematic("statError","Nom","dummy","dummy","tree","shapeStat","Gaussian")`
`fakes.setStatConfig(False)`
`fakes.addSystematic(statSys)`

Does this explanation make sense to you? And can you confirm, that the proposed solution is correct?

Thanks,
Jakob

**Reply by luilongo** [reply]
Hi Jakob,
the gamma parameter coming from `setStatConfig` is not set to constant (it's in any case included in the fit) but I believe you are right for the rest. Your fake stat error is summed up with the ttbar one and just one gamma parameter is used; in such way we lose somehow the correspondance with the rescaled sample.
I believe your solution is the correct one. Could you send me the 2 outputs with `setStatConfig` and the use of `shapeStat` with the yields of the 2 cases?
Cheers,
Luigi

**Reply by jakobnov** [reply]
Hi Luigi,

yes, gamma parameter is floating in either case. I just meant that the standard deviation of the gamma parameter is kept constant regardless of the nuisance parameter values. I attach outputs for both cases, the name of the `shapeStat` systematic is `statError`:

[shapeStat.log](https://github.com/histfitter/histfitter/files/7169974/shapeStat.log)
[statConfig.log](https://github.com/histfitter/histfitter/files/7169983/statConfig.log)


Thanks,
Jakob
